packages/material-ui/src/Switch/Switch.js

```diff
@@ -89,17 +85,13 @@ export const styles = (theme) => ({
           backgroundColor: 'transparent',
         },
       },
-    },
-    '&$disabled': {
-      color: theme.palette.mode === 'light' ? theme.palette.grey[400] : theme.palette.grey[800],
+      '&$disabled': {
+        color: lighten(theme.palette.secondary.main, 0.62),
```

Comment:
We are trying to limit the usage of color transformation in the components to add CSS variables support. Do you think that it's possible to implement the design with an opacity?

Reply:
I see; I think implementing this with opacity might be complicated. I started out by trying to use opacity, but discovered because of the track underneath the thumb, setting opacity less than 1 to get a lighter color causes the track to become visible under the thumb. That's why I went for color transformation instead so that the thumb doesn't become semi-transparent and reveal the track. If done using opacity, I think there would need to be an additional element underneath the thumb that's solid in color to prevent the track from becoming visible when the thumb's opacity is less than 1. If that's preferable, let me know and I can give it a shot. Also open to other suggestions!

Reply:
Ok, thanks for sharing the details of your exploration.

Reply:
It does sound like a tricky issue. We can focus on matching the design for now